src/common/src/types/interval.rs

```diff
@@ -1007,6 +1010,78 @@ impl Interval {
         };
         format!("P{years}Y{months}M{days}DT{hours}H{minutes}M{seconds}{fract_str}S")
     }
+
+    pub fn from_iso_8601(s: &str) -> Result<Self> {
```

Comment:
This function is quite verbose, can we make it concise a bit?

Reply:
And why did you remove the comment for this function?

Reply:
FYI `caps.get(x).unwrap().as_str()` is equivalent to `caps[x]`. The repeated error handling code can also be simplified.

Reply:
I tried to wrap it into a closure, so that I could use `map_err` at the very end.
however, for `Option::None`, I still have to use `context` to convert it to `Result::Error`...
comments are also added back :)

Reply:
Good idea..
but how to return an `Option` early if it is `None`? We can use `?` for `Result` but not for an `Option`...

Reply:
`?` works for both `Result` and `Option`: [rustdoc](https://doc.rust-lang.org/std/option/#the-question-mark-operator-)

Reply:
I will fix it in this pr 🙂
https://github.com/risingwavelabs/risingwave/pull/9952